### DISCUSSION TITLE
How to show feedback

### DISCUSSION THREAD

**Discussion by rogerhutchings** [zooniverse/front-end-monorepo#2433] [General]
@eatyourgreens and I were talking about this on Wednesday, and he suggested showing the feedback on the classifier itself, rather than in a separate modal (which testers found too small). This is how PH used to show its feedback - I think it was Gravity Spy that started the whole feedback in a modal thing.

Maybe this is something we should A/B test in the future?

**Comment by srallen** [comment]
Gravity Spy does not show its feedback in a modal; It was suggested by me when the generalized feedback was implemented to avoid the issue of the Done & Talk link and users missing the feedback message if they clicked it. The possibility that users miss Gravity Spy's feedback remains and one that the project has accepted because it is in the task area and we do not want to leak that the subject is a gold standard subject, so we continue to show the Done & Talk link. 

If we spend our time A/B splitting something, I'd rather A/B split the removal of the Done & Talk link from the task area. If we can prove that Talk engagement does not decrease with the Talk link somewhere else in the classifier area, then how feedback shows becomes a non-issue in the task area.

**Comment by srallen** [comment]
I'll also use this discussion to add the comments about setup and usability:

- Let's leave the customization of messaging via subject metadata undocumented. This option has been confusing for project builders
- If a project has many rules, but the default messages are the same, should be reduce the message somehow?
   - Should this only apply to projects with drawing tools?

**Comment by eatyourgreens** [comment]
Maybe drawing task rules should reduce down to one message per drawing tool. 100 messages for 100 penguins on Penguin Watch would be annoying. Messages for the marks would still have to be generated by the strategies for each tool (point, column etc.)

**Comment by eatyourgreens** [comment]
In PFE, each task has a summary method so feedback could build on the summary by adding info about how correct/incorrect it was.

**Comment by camallen** [comment]
> Maybe drawing task rules should reduce down to one message per drawing tool. 100 messages for 100 penguins on Penguin Watch would be annoying. Messages for the marks would still have to be generated by the strategies for each tool (point, column etc.)

I like this suggestion, if we can't get that done in time for the Tess relaunch, can we drop the text feedback entirely? 

That project has visual feedback and it'll be the only project on the rebuilt classifier for a while.

**Comment by srallen** [comment]
> can we drop the text feedback entirely?

I disagree with this as I would strongly bet we would get even more complaints about a random modal with no explanation given as to why it is displaying. It's also bad for accessibility. 

The path of least resistance right now is writing a reducer, however, I really want us to be capable of making hard deadlines for launches and deciding what's critical and what's not.
